From 3ea76e83c0443ee7f9368a65898b76e0577d6b39 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 13 Dec 2011 17:18:13 +0100
Subject: [PATCH] Fix AAAA removal update

Port for 0.65 from:
    r819 | olaf | 2009-11-30 10:04:15 +0100 (Po, 30 lis 2009) | 1 line
    Dynamic update error fixed AAAA

    r831 | olaf | 2009-12-28 16:57:43 +0100 (Po, 28 pro 2009) | 1 line
    Fix rt.cpan.org 43770 rr_del broken for AAAA, DHCID, and HINFO

and subsequent regression fix:
    r833 | olaf | 2009-12-28 17:25:09 +0100 (Po, 28 pro 2009) | 1 line
    Minor HINFO Bug just introduced, now repaired

Petr Pisar: Changelog and some white space changes removed.
---
 lib/Net/DNS/RR/AAAA.pm  |    5 +++--
 lib/Net/DNS/RR/HINFO.pm |    3 ++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/lib/Net/DNS/RR/AAAA.pm b/lib/Net/DNS/RR/AAAA.pm
index 1a0d10e..af1cfe6 100644
--- a/lib/Net/DNS/RR/AAAA.pm
+++ b/lib/Net/DNS/RR/AAAA.pm
@@ -24,7 +24,7 @@ sub new {
 
 sub new_from_string {
 	my ($class, $self, $string) = @_;
-	$self->{"address"}=$string;
+	$self->{"address"}=$string if $string;
 	bless $self, $class;
 	return $self->_normalize_AAAA();
 }
@@ -38,8 +38,8 @@ sub rdatastr {
 sub rr_rdata {
 	my $self = shift;
 	my $rdata = "";
-	$self->_normalize_AAAA();
 	if (exists $self->{"address"}) {
+		$self->_normalize_AAAA();
 		my @addr = split(/:/, $self->{"address"});
 		$rdata .= pack("n8", map { hex $_ } @addr);
 	}
@@ -52,6 +52,7 @@ sub rr_rdata {
 
 sub _normalize_AAAA {
 	my $self=shift();
+	return $self unless exists $self->{"address"};
 	return $self->{"address"} if $self->{normalized};
 	
 	my $string=$self->{"address"};
diff --git a/lib/Net/DNS/RR/HINFO.pm b/lib/Net/DNS/RR/HINFO.pm
index b062a43..eac4d36 100644
--- a/lib/Net/DNS/RR/HINFO.pm
+++ b/lib/Net/DNS/RR/HINFO.pm
@@ -40,6 +40,7 @@ sub new_from_string {
 	
 	bless $self, $class;
         
+        return $self unless $rdata_string;
 	$self->_build_char_str_list($rdata_string);
 	my @elements= $self->char_str_list();
 	if (@elements==2){
@@ -52,7 +53,7 @@ sub new_from_string {
 	}
 	
 	
-	return bless $self, $class;
+	return $self;
 }
 
 sub rdatastr {
-- 
1.7.7.4

